kornia/augmentation/container/ops.py

```diff
@@ -195,32 +197,85 @@ def transform(cls, input: Tensor, module: Module, param: ParamItem, extra_args:
             param: the corresponding parameters to the module.
         """
         if isinstance(module, (K.GeometricAugmentationBase2D,)):
-            input = module.transform_masks(
-                input,
-                params=cls.get_instance_module_param(param),
-                flags=module.flags,
-                transform=module.transform_matrix,
-                **extra_args,
-            )
+            if isinstance(input, list):
+                tfm_input = []
+                for inp in input:
+                    inp = module.transform_tensor(inp)
+                    tfm_inp = module.transform_masks(
+                        inp,
+                        params=cls.get_instance_module_param(param),
+                        flags=module.flags,
+                        transform=module.transform_matrix,
+                        **extra_args,
+                    )
+                    tfm_input.append(tfm_inp)
+                input = tfm_input
+
+            else:
+                input = module.transform_masks(
+                    input,
+                    params=cls.get_instance_module_param(param),
+                    flags=module.flags,
+                    transform=module.transform_matrix,
+                    **extra_args,
+                )
 
         elif isinstance(module, (K.GeometricAugmentationBase3D,)):
             raise NotImplementedError(
                 "The support for 3d mask operations are not yet supported. You are welcome to file a PR in our repo."
             )
 
         elif isinstance(module, (_AugmentationBase)):
-            input = module.transform_masks(
-                input, params=cls.get_instance_module_param(param), flags=module.flags, **extra_args
-            )
+            if isinstance(input, list):
+                tfm_input = []
+                for inp in input:
+                    inp = module.transform_tensor(inp)
+                    tfm_inp = module.transform_masks(
+                        inp, params=cls.get_instance_module_param(param), flags=module.flags, **extra_args
```

Comment:
The params are shared across different masks in the list I guess. So, maybe it is better to retrieve the params only once.

Reply:
Done